compiler/rustc_mir_transform/src/check_alignment.rs

```diff
@@ -87,6 +84,29 @@ fn insert_alignment_check<'tcx>(
         ))),
     });
 
+    // If this target does not have reliable alignment, further limit the mask.
+    #[allow(irrefutable_let_patterns)]
+    if let max_align = tcx.sess.target.max_reliable_alignment()
+        && max_align < Align::MAX
+    {
+        let max_mask = max_align.bytes() - 1;
+        let max_mask = Operand::Constant(Box::new(ConstOperand {
+            span: source_info.span,
+            user_ty: None,
+            const_: Const::Val(
+                ConstValue::Scalar(Scalar::from_target_usize(max_mask, &tcx)),
+                tcx.types.usize,
+            ),
+        }));
+        stmts.push(Statement {
+            source_info,
+            kind: StatementKind::Assign(Box::new((
+                alignment_mask,
+                Rvalue::BinaryOp(BinOp::Sub, Box::new((Operand::Copy(alignment_mask), max_mask))),
+            ))),
+        });
```

Comment:
This kind of bit twiddling hurts my brain. Why are we subtracting masks? I think of masks in terms of BitAnd/BitOr.

Reply:
Yeah that's a copy-paste mistake :joy: I meant BitAnd...